app/models/immunity_detail.rb

```diff
@@ -49,4 +48,30 @@ def accepted?
   def current_review_immunity_detail
     review_immunity_details.last
   end
+
+  def start_date
+    evidence_groups.order(start_date: :asc).first.start_date
+  end
+
+  def end_date
+    group = evidence_groups.order(end_date: :asc, start_date: :asc).last
+    group.end_date || group.start_date
+  end
+
+  def evidence_gaps? # rubocop:disable Metrics/AbcSize
+    return if evidence_groups.blank?
+
+    # https://stackoverflow.com/a/8765467
+    date_ranges = evidence_groups.map do |g|
+      Range.new(g.start_date.to_datetime.to_date, g.end_date.to_datetime.to_date)
```

Comment:
I've just pulled this locally, and it's occasionally blowing up if there isn't a start date or an end date present, which is possible

Reply:
(think this also applies in other places in the code where it's assuming they are both present)

Reply:
Sorry, one more comment. I had to do something like this in my PR, can't you just do 
```    
evidence_groups.any?(&:missing_evidence)
```
because the assessor marks whether there are any missing gaps, I don't think the code has to be clever enough to work it out

Reply:
Ah that's a good point, I'm not sure how this should account for missing dates. Exclude them? Or set the end date to today?

Reply:
So turns out our cards cover similar things... 😬 I did 
```
      <% if immunity_detail.earliest_evidence_cover.present? && immunity_detail.latest_evidence_cover.present? %>
        from <%= immunity_detail.earliest_evidence_cover %> to <%= immunity_detail.latest_evidence_cover %>
      <% else %>
        <%= immunity_detail.earliest_evidence_cover || immunity_detail.latest_evidence_cover %>
      <% end %>
```
but not sure if that covers it all

Reply:
> `evidence_groups.any?(&:missing_evidence)`

I'm not sure if that would be enough — it seems to me that it's possible for all of the evidence groups to be complete for the range each one covers separately, _but_ for there to be too few evidence groups to cover the total range. But if we think that's sufficient then we should definitely go with the simpler code.